pyproject.toml

```diff
@@ -38,7 +38,7 @@ types-pytz = ">= 2022.1.1"
 mypy = "0.990"
 pyarrow = ">=10.0.1"
 pytest = ">=7.1.2"
-pyright = ">=1.1.281"
+pyright = "1.1.282"
```

Comment:
Might be nice to add an inline TODO/comment citing the issue. Would also be fine merging it without it - but would at least wait until you get an initial response on the pyright issue (if it is going to be fixed soon, we could, for example, change the constraint to != 1.1.283).